contribute/design/1.x/apis/cstorvolume_policy.md

```diff
@@ -0,0 +1,318 @@
+
+	// Luworkers sets the number of threads that are working on above queue
+	LuWorkers int64 `json:"luWorkers,omitempty"`
+
+	// Monitor enables or disables the target exporter sidecar
+	Monitor bool `json:"monitor,omitempty"`
+
+	// ReplicationFactor represents maximum number of replicas
+	// that are allowed to connect to the target
+	ReplicationFactor int64 `json:"replicationFactor,omitempty"`
+
+	// Resources are the compute resources required by the cstor-target
+	// container.
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// AuxResources are the compute resources required by the cstor-target pod
+	// side car containers.
+	AuxResources *corev1.ResourceRequirements `json:"auxResources,omitempty"`
+
+	// Tolerations, if specified, are the target pod's tolerations
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// PodAffinity if specified, are the target pod's affinities
+	PodAffinity *corev1.PodAffinity `json:"affinity,omitempty"`
+
+	// NodeSelector is the labels that will be used to select
+	// a node for target pod scheduleing
+	// Required field
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// PriorityClassName if specified applies to this target pod
+	// If left empty, no priority class is applied.
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+}
+
+// ReplicaSpec represents configuration related to replicas resources
+type ReplicaSpec struct {
+	// ZvolWorkers represents number of threads that executes client IOs
+	ZvolWorkers *int32 `json:"zvolWorkers,omitempty"`
+}
+
+// Provision represents volume provisioning configuration
+type Provision struct {
+	// replicaAffinity is set to true then volume replica resources need to be
+	// distributed across the cstor pool instances based on the given topology
+	ReplicaAffinity bool `json:"replicaAffinity"`
+}
+
+// ReplicaPoolSpec represents the volume replicas pool information
+type ReplicaPoolSpec struct {
+	// PoolInfo represents the pool information of replicas
+	PoolInfo []ReplicaPoolInfo `json:"poolInfo,omitempty"`
+}
+
+// ReplicaPoolInfo represents the pool information of volume replica
+type ReplicaPoolInfo struct {
+	// PoolName represents the pool name where volume replica exists
+	PoolName string `json:"poolName,omitempty"`
+	//TODO: UID also can be added
+}
+
+// CStorVolumePolicyStatus is for handling status of CstorVolumePolicy
+type CStorVolumePolicyStatus struct {
+	Phase string `json:"phase,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// CStorVolumePolicyList is a list of CStorVolumePolicy resources
+type CStorVolumePolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []CStorVolumePolicy `json:"items"`
+}
+```
+
+# Changes to existing types
+
+- Added `ReplicaPoolSpec` in `VolumePolicy` schema represents pool related information required
```

Comment:
should ReplicaPoolSpec be part of CStorVolumeConfig instead of CStorVolumePolicy?

Reply:
its `ReplicaPoolInfo` now. 
We had discussed this earlier and thought of using as policy in case user give the set of pools to provision the volumes like a StorageClass.